yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -271,15 +271,20 @@ func assessMigration() (err error) {
 
 		// Handle replica discovery and validation (PostgreSQL only)
 		if source.DBType == POSTGRESQL {
-			validatedReplicaEndpoints, err = handleReplicaDiscoveryAndValidation(sourceReadReplicaEndpoints)
+			validatedReplicaEndpoints, err = migassessment.HandleReplicaDiscoveryAndValidation(source.DB().(*srcdb.PostgreSQL), sourceReadReplicaEndpoints)
```

Comment:
i would suggest pass sourceDB here. 
Do the type check and whatever you need to do inside that function.